src/SdkCommon/ClientRuntime.Azure/ClientRuntime.Azure/AzureClientExtensions.cs

```diff
@@ -349,7 +349,7 @@ public static partial class AzureClientExtensions
             var statusCode = initialResponse.Response.StatusCode;
             var method = initialResponse.Request.Method;
             if (statusCode == HttpStatusCode.OK || statusCode == HttpStatusCode.Accepted ||
-                (statusCode == HttpStatusCode.Created && method == HttpMethod.Put) ||
+                (statusCode == HttpStatusCode.Created && (method == HttpMethod.Put || method == new HttpMethod("PATCH"))) ||
```

Comment:
Hi @shahabhijeet , this change is required for the EventGrid client ( https://github.com/Azure/azure-sdk-for-net/pull/3600 )

Did you want me to write up some tests for this, or should I close this ahead of Microsoft internally doing the work?